store/store.go

```diff
@@ -531,11 +531,42 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
+type metadataInput struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision"`
+	Epoch    string `json:"epoch"`
+	// FIXME: not in snapinfo yet
+	//Confinement string `json:"confinement"`
+}
+
+type metadataWrapper struct {
+	Snaps  []metadataInput `json:"snaps"`
+	Fields []string        `json:"fields"`
+}
+
 // Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+func (s *SnapUbuntuStoreRepository) Updates(installed []*snap.Info, auther Authenticator) (snaps []*snap.Info, err error) {
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+	// build input for the updates endpoint
+	is := make([]metadataInput, 0, len(installed))
+	for _, inst := range installed {
+		if inst.Revision.Local() {
+			continue
+		}
+		is = append(is, metadataInput{
+			SnapID:   inst.SnapID,
+			Channel:  inst.Channel,
```

Comment:
@pedronis the current code in the daemons gets all the info via `allLocalSnapInfos()`. Which will use snapstate.Current() to populate this info.